taui/src/actions/profile.js

```diff
@@ -29,6 +29,35 @@ export const setProfile = (profile) => (dispatch, getState) => {
   }
 }
 
+export const saveProfile = (profile, authToken) => (dispatch, getState) => {
+  axios.put(`/api/user/`, profile, {
```

Comment:
To expand on the earlier comment about this not returning anything, this function can be summarized as:
```javascript
saveProfile = (profile, authToken) => (dispatch, getState) => {
  axios.put().then().catch()
}
```
Which shows that it's not returning anything because it only contains one expression, and that expression isn't returned. The `return` inside the `catch()` function will only affect the value of the Promise that is returned by `catch()` -- [`catch()` always returns a Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch), and the return value of any function that has been passed to `catch()` as the `onRejected` parameter is used to resolve the Promise.